File: leetcode/0001/twoSum.py
# -*- coding: utf-8 -*-

def twoSum(nums, target):
    hashTable={}
    for i,num in enumerate(nums):
        if hashTable.get(target - num) is not None:
            return [hashTable.get(target - num), i]
        hashTable[num] = i

result = twoSum([2,7,11,15], 9)
print(result)

File: leetcode/0001/twoSum.swift
func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
    var hash: Dictionary<Int, Int> = [:]
    for (i, num) in nums.enumerated() {
        if let lastIndex = hash[target - num] {
            return [lastIndex, i]
        }
        hash[num] = i
    }
    return []
}

let result = twoSum([2,7,11,15], 9)
print(result)
